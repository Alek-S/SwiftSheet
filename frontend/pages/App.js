import React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router-dom';

import Header from '../components/Header/Header';
import Footer from '../components/Footer/Footer';
import FrontPage from './FrontPage/FrontPage';
import UploadPage from './UploadPage/UploadPage';
import ViewPage from './ViewPage/ViewPage';
import SheetPage from './SheetPage/SheetPage';
import ScrollToTop from '../components/ScrollToTop/ScrollToTop';
import PrivacyPage from './PrivacyPage/PrivacyPage';

/** @function
 * @name App
 *
 * @returns {JSX}
 */
const App = ({ match }) => {
	return (
		<ScrollToTop>
			<Header />
			<Switch>
				<Route exact path="/" component={FrontPage} />
				<Route path="/privacy" component={PrivacyPage} />
				<Route path="/upload" component={UploadPage} />
				<Route path="/view" component={ViewPage} />
				<Route path="/:sheetId" component={SheetPage} />
			</Switch>
			<Footer />
		</ScrollToTop>
	);
};

export default hot(module)(App);
