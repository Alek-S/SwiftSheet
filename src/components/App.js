import React from 'react';
import styled from 'styled-components';

import Header from './Header/Header';
import Footer from './Footer/Footer';

/** @class
 * @name App
 *
 * @returns {JSX}
 */
export default class App extends React.Component {
	render() {
		return (
			<div>
				<Header />
				<Body />
				<Footer />
			</div>
		);
	}
}

//TODO: remove Body - placeholder to test header/footer positions
/* istanbul ignore next*/
const Body = styled.div`
	background-color: ${props => props.theme.color.background};
	height: 1500px;
`;
