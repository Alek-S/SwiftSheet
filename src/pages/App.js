import React from 'react';
import styled from 'styled-components';
import { hot } from 'react-hot-loader';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import Header from '../components/Header/Header';
import Footer from '../components/Footer/Footer';
import Filedrop from '../components/Filedrop/Filedrop';

/** @function
 * @name App
 *
 * @returns {JSX}
 */
const App = () => {
	return (
		<BrowserRouter>
			<div>
				<Header />
				<Switch>
					<Route exact path="/" component={Filedrop} />
				</Switch>
				<Body />
				<Footer />
			</div>
		</BrowserRouter>
	);
};

//TODO: remove Body - placeholder to test header/footer positions
/* istanbul ignore next*/
const Body = styled.div`
	background-color: ${props => props.theme.color.background};
	height: 1500px;
`;

export default hot(module)(App);
