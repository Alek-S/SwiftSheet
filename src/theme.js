/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
