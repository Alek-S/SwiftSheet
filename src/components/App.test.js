import React from 'react';
import Enzyme, { configure, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import App from './App.js';
import theme from '../theme';

Enzyme.configure({ adapter: new Adapter() });

const shallowWithTheme = (tree, theme) => {
	const context = shallow(<ThemeProvider theme={theme} />)
		.instance()
		.getChildContext();
	return shallow(tree, { context });
};

describe('App', () => {
	it('matches the snapshot', () => {
		const tree = shallowWithTheme(<App />, theme);
		expect(toJson(tree)).toMatchSnapshot();
	});

	it('contains a Header', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('Header').length).toBe(1);
	});

	it('contains a Footer', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('Footer').length).toBe(1);
	});
});
