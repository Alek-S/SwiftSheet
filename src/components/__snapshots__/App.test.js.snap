// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div>
  <Header />
  <App__Body />
  <Footer />
</div>
`;
